WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/RegisterDomainCoordinator.swift

```diff
@@ -0,0 +1,195 @@
+import Foundation
+
+class RegisterDomainCoordinator {
+
+    // MARK: Type Aliases
+
+    typealias DomainPurchasedCallback = ((UIViewController, String) -> Void)
+    typealias DomainAddedToCartCallback = ((UIViewController, String) -> Void)
+
+    // MARK: Variables
+
+    var site: Blog?
+    var domainPurchasedCallback: DomainPurchasedCallback?
+    var domainAddedToCartCallback: DomainAddedToCartCallback?
+    var domain: FullyQuotedDomainSuggestion?
+
+    private var webViewURLChangeObservation: NSKeyValueObservation?
+
+    init(site: Blog?, domainPurchasedCallback: RegisterDomainCoordinator.DomainPurchasedCallback? = nil) {
+        self.site = site
+        self.domainPurchasedCallback = domainPurchasedCallback
+    }
+
+    // MARK: Public Functions
+
+    func createCart(_ domain: FullyQuotedDomainSuggestion,
+                    onSuccess: @escaping () -> (),
+                    onFailure: @escaping () -> ()) {
+        let siteID = site?.dotComID?.intValue
+        let proxy = RegisterDomainDetailsServiceProxy()
+        proxy.createPersistentDomainShoppingCart(siteID: siteID,
+                                                 domainSuggestion: domain.remoteSuggestion(),
+                                                 privacyProtectionEnabled: domain.supportsPrivacy ?? false,
+                                                 success: { _ in
+            onSuccess()
+        },
+                                                 failure: { _ in
+            onFailure()
+        })
+    }
+
+    func presentWebViewForNoSite(on viewController: UIViewController,
+                                 domainSuggestion: FullyQuotedDomainSuggestion) {
+        guard let url = URL(string: Constants.noSiteCheckoutWebAddress) else {
+            return
+        }
+
+        presentCheckoutWebview(on: viewController,
+                               domainSuggestion: domainSuggestion,
+                               url: url,
+                               title: TextContent.checkoutTitle,
+                               shouldPush: true)
+    }
+
+    func presentWebViewForCurrentSite(on viewController: UIViewController,
+                                              domainSuggestion: FullyQuotedDomainSuggestion) {
+        guard let site,
+              let homeURL = site.homeURL,
+              let siteUrl = URL(string: homeURL as String), let host = siteUrl.host,
+              let url = URL(string: Constants.checkoutWebAddress + host) else {
+            return
+        }
```

Comment:
[Suggestion] If one of those constants fail, I think that should be considered an unexpected error and worth catching on Sentry. I had a similar implementation in `SiteCreationPurchasingWebFlowController.swift` and there was few cases where one of these constants was `nil`.

Reply:
Nice suggestion! Addressed in 243562a5ea5da815dd5354a4fd0ad3dd5804fd02 👍